rmutil/heap.h

```diff
@@ -0,0 +1,38 @@
+#ifndef __HEAP_H__
+#define __HEAP_H__
+
+#include "vector.h"
+
+
+/* Make heap from range
+ * Rearranges the elements in the range [first,last) in such a way that they form a heap.
+ * A heap is a way to organize the elements of a range that allows for fast retrieval of the element with the highest
+ * value at any moment (with pop_heap), even repeatedly, while allowing for fast insertion of new elements (with
+ * push_heap).
+ * The element with the highest value is always pointed by first. The order of the other elements depends on the
+ * particular implementation, but it is consistent throughout all heap-related functions of this header.
+ * The elements are compared using cmp.
+ */
+void make_heap(Vector *v, size_t first, size_t last, int (*cmp)(char *, char *));
+
+
+/* Push element into heap range
+ * Given a heap in the range [first,last-1), this function extends the range considered a heap to [first,last) by
+ * placing the value in (last-1) into its corresponding location within it.
+ * A range can be organized into a heap by calling make_heap. After that, its heap properties are preserved if elements
+ * are added and removed from it using push_heap and pop_heap, respectively.
+ */
+void push_heap(Vector *v, size_t first, size_t last, int (*cmp)(char *, char *));
```

Comment:
I keep (make_heap, push_heap, pop_heap) interface here very close to c++ (std::make_heap, std::push_heap, std::pop_heap). I think I can write something like std::priority_queue to wrap the cmp function and underlying Vector together, which would provide a cleaner interface.